python/pyarrow/_flight.pyx

```diff
@@ -721,6 +796,28 @@ cdef class SchemaResult(_Weakrefable):
         check_flight_status(self.result.get().GetSchema(&dummy_memo).Value(&schema))
         return pyarrow_wrap_schema(schema)
 
+    def serialize(self):
+        """Get the wire-format representation of this type.
+
+        Useful when interoperating with non-Flight systems (e.g. REST
+        services) that may want to return Flight types.
+
+        """
+        return GetResultValue(self.result.get().SerializeToString())
+
+    @classmethod
+    def deserialize(cls, serialized):
+        """Parse the wire-format representation of this type.
+
+        Useful when interoperating with non-Flight systems (e.g. REST
+        services) that may want to return Flight types.
+
+        """
+        # cdef SchemaResult result = SchemaResult.__new__(SchemaResult)
+        # result.result.reset(new CSchemaResult(GetResultValue(
+        #     CSchemaResult.Deserialize(tobytes(serialized)))))
+        # return result
```

Comment:
Commented code?

Reply:
Fixed, add default constructor for SchemaResult on C++